lib/health.js

```diff
@@ -48,17 +48,37 @@ class HealthCheck {
     };
   }  
 
+  http() {
+    return (request, response) => {
+      const requestPath = request.url;
+      if (requestPath !== this.getHealthUri()) return false;
+
+      const self = this;
+
+      (async function () {
+        const responseContent = await self.healthResponse(self.opts);
+        const httpCode = (response.status === STATUS_ERR ) ? 503 : 200;
+        
+        response.writeHead(httpCode, { 'Content-Type': 'application/health+json; charset=utf-8' });
+        response.end(JSON.stringify(responseContent));
+      })();
+
+      return true;
+    };
+  }
+
   addCheck (componentName, metricName, checkerPromise) {
     const key = `${componentName}:${metricName}`;
-    this.checks[key] = checkerPromise;
+    this.checks[key] = {};
```

Comment:
Should there be some sort of warning if you are adding a duplicate check?  Here you are just wiping out a check if it already exists.

Reply:
Touche! Fixed.

Reply:
You mean - if a check *with the same key* already exists? I assume if you call addCheck() with the same key - then you do indeed want to replace the old check? Not a fair assumption?

Reply:
`addCheck` should throw instead of overwriting, while (a new) `setCheck` should allow overwriting without complaint?

Reply:
ok, will incorporate that in the next PR